src/js/administration/components/Api.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { useMutation, useQueryClient } from "react-query";
+import { ExternalLink, LoadingPlaceholder } from "../../base";
+import { updateSettings } from "../api";
+import { settingsQueryKeys, useFetchSettings } from "../querys";
+import { SettingsCheckbox } from "./SettingsCheckbox";
+
+/**
+ * Displays the API settings and allows the users to toggle API access for clients
+ */
+export default function Api() {
+    const { data, isLoading } = useFetchSettings();
+    const queryClient = useQueryClient();
+    const mutation = useMutation(updateSettings, {
+        onSuccess: () => {
+            queryClient.invalidateQueries(settingsQueryKeys.all());
+        },
+    });
+
+    if (isLoading) {
+        return <LoadingPlaceholder />;
+    }
```

Comment:
How does this look visually in practice? (make sure you check with some latency as well) I get the feeling the loader will look pretty bad here. Might sense to prefetch in the component above and block loading of both instanceMessage and API until the cache is populated